bgpd/bgp_routemap.c

```diff
@@ -5231,6 +5282,43 @@ DEFUN_YANG (no_set_distance,
 	return nb_cli_apply_changes(vty, NULL);
 }
 
+DEFUN_YANG(set_l3vpn_nexthop_encapsulation, set_l3vpn_nexthop_encapsulation_cmd,
+	   "[no] set l3vpn next-hop encapsulation gre",
+	   NO_STR SET_STR
+	   "L3VPN operations\n"
+	   "Next hop Information\n"
+	   "Encapsulation options (for BGP only)\n"
+	   "Accept L3VPN traffic over GRE encapsulation\n")
+{
+	const char *xpath =
+		"./set-action[action='frr-bgp-route-map:set-l3vpn-nexthop-encapsulation']";
+	char xpath_value[XPATH_MAXLEN];
+	enum nb_operation operation;
+	int ret;
+
+	if (strmatch(argv[0]->text, "no"))
+		operation = NB_OP_DESTROY;
+	else
+		operation = NB_OP_CREATE;
+
+	nb_cli_enqueue_change(vty, xpath, operation, NULL);
+	if (operation == NB_OP_DESTROY)
+		return nb_cli_apply_changes(vty, NULL);
+
+	ret = snprintf(
+		xpath_value, sizeof(xpath_value),
+		"%s/rmap-set-action/frr-bgp-route-map:l3vpn-nexthop-encapsulation",
+		xpath);
```

Comment:
Looking at this - I don't see `xpath` ever being modified, indeed it's `const`. So why do we need an `snprintf` here at all? Can't you hardcode this entire path? Saves us all this printing and checking code.

Reply:
simplified code. @qlyoung , please re review